tests/Doctrine/Tests/ORM/Functional/EnumTest.php

```diff
@@ -260,6 +260,86 @@ public function testEnumArrayInDtoHydration(): void
         self::assertEqualsCanonicalizing([Unit::Gram, Unit::Meter], $result[0]->supportedUnits);
     }
 
+    public function testEnumSingleEntityChangeSetsSimpleObjectHydrator(): void
+    {
+        $this->setUpEntitySchema([Card::class]);
+
+        $card       = new Card();
+        $card->suit = Suit::Clubs;
+
+        $this->_em->persist($card);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $result = $this->_em->find(Card::class, $card->id);
+
+        $this->_em->getUnitOfWork()->recomputeSingleEntityChangeSet(
+            $this->_em->getClassMetadata(Card::class),
+            $result
+        );
+
+        self::assertFalse($this->_em->getUnitOfWork()->isScheduledForUpdate($result));
+
+        $result->suit = Suit::Hearts;
+
+        $this->_em->getUnitOfWork()->recomputeSingleEntityChangeSet(
+            $this->_em->getClassMetadata(Card::class),
+            $result
+        );
+
+        self::assertTrue($this->_em->getUnitOfWork()->isScheduledForUpdate($result));
+    }
+
+    public function testEnumSingleEntityChangeSetsObjectHydrator(): void
+    {
+        $this->setUpEntitySchema([Card::class]);
+
+        $card       = new Card();
+        $card->suit = Suit::Clubs;
+
+        $this->_em->persist($card);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $result = $this->_em->createQueryBuilder()
+            ->from(Card::class, 'c')
+            ->select('c')
+            ->getQuery()
+            ->getResult();
+
+        $this->_em->getUnitOfWork()->recomputeSingleEntityChangeSet(
+            $this->_em->getClassMetadata(Card::class),
+            $result[0]
+        );
+
+        self::assertFalse($this->_em->getUnitOfWork()->isScheduledForUpdate($result[0]));
+    }
+
+    public function testEnumArraySingeEntityChangeSets(): void
```

Comment:
:monkey: spotted (:fr: joke)
```suggestion
    public function testEnumArraySingleEntityChangeSets(): void
```